README.md

```diff
@@ -45,26 +96,53 @@ policy = jsonencode({
       }
     ]
   })
-
 ```
 
-You can then add tags to your role:  
+You can add tags to your role using Terraform.  eg;
 ```terraform
   tags = {
     secrets_default = "${var.service_name}/${var.short_env}/secrets"
   }
 ```
 
+---
+
+## Building, testing and linting
+
+Build the binary using:  
+```shell
+make artifact
+```
+The resulting binary will be called `fetch-secrets`.
+
+_NOTE: If you need to compile a linux specific binary run and you're on a Mac use:_
+```shell
+GOOS=linux go build
+```
+
+Test the code using:
+```shell
+make test
+```
+
+Lint the code using:
+```shell
+make lint
+```
+
+---
 
-#### Building
+## Troubleshooting
 
-Run the following commands:  
-```sh
-go get
-go build
-````
+`fetch-secrets` uses JSON logging throughout.  The logging should indicate, the role determined, the tag keys and values as well as the secret-names to fetch values for.  
+If logs are unavailable, the following exit-codes should be helpful in indicating where the issue might be:
 
-The resulting binary will be in this directory called `fetch-secrets`.
+| Exit Code | Description                             |
```

Comment:
If we're documenting these here it would probably make sense for them to be variables/types instead of literals in the code.